lesson-8/script.js

```diff
@@ -0,0 +1,177 @@
+const isNumber = (number) => {
+    return typeof number === 'number' && !isNaN (number) 
+}
+const isString = (string) => {
+    return typeof string === 'string' && string.length > 0  
+}
+
+// 1. Створити масив, довжину та елементи якого задає користувач (через prompt). Елементами масиву повинні бути числа
+
+const createNewArray = () => {
+    let newArray = [];
+    const askNewArrayLength = +prompt("How many elements would array contain?");
+    if (!isNumber(askNewArrayLength)) {
+        console.log(`You must enter a number.`);
+        return
+    } 
+    
+    for (let i = 0; i <= askNewArrayLength - 1; i++) {
+        let askNewArrayElements = +prompt(`Enter value of the element #${i}.`);
+        if (!isNumber(askNewArrayElements)) {
+            console.log(`Your array values can be only a numbers.`);
+            break
+        } 
+        newArray[i] = askNewArrayElements
+    }
+
+    if (newArray.length > 0) {
+        return newArray;
+    }
+}
+const newArray = createNewArray()
+
+// 2. Відсортувати масив за зростанням.
+
+newArray.sort((a, b) => a - b);
+
+// 3. Дано масив - список покупок. Кожен елемент масиву - це обʼєкт вигляду: {productName: 'bread', productPrice: 14.5, productQuantity: 2}. Мінімальний довжина такого масиву - 6
+
+let purchaseList = [
+    {productName: 'bread', productPrice: 14.5, productQuantity: 1},
+    {productName: 'mineral water', productPrice: 11.2, productQuantity: 6},
+    {productName: 'sour cream', productPrice: 24, productQuantity: 2},
+    {productName: 'carrot', productPrice: 17.9, productQuantity: 3},
+    {productName: 'energy bar', productPrice: 4.5, productQuantity: 7},
+    {productName: 'milk', productPrice: 35, productQuantity: 3},
+]
+
+// Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+
+let totalPurchaseSum = 0;
+purchaseList.forEach(product => totalPurchaseSum += product.productPrice * product.productQuantity)
```

Comment:
As an option reduce can be used and nice to have at least console.log for understanding what exact value was calculated

`console.log(`Total cost: ${totalPurchaseSum}`);`